aten/src/ATen/native/cpu/DistributionTemplates.h

```diff
@@ -147,6 +147,50 @@ static void normal_fill_16(scalar_t *data, const scalar_t mean, const scalar_t s
     data[j + 8] = radius * std::sin(theta) * std + mean;
   }
 }
+ 
+#if defined(__VSX__)  || defined(CPU_CAPABILITY_VSX)
+static void normal_fill_16_VSX(float *data,Vectorized<float> &two_pi,Vectorized<float> &one,Vectorized<float> &minus_two, Vectorized<float> &mean,Vectorized<float> &std) {
+  using Vec = Vectorized<float>;
+  Vec u1=one-Vec::loadu(data);// [0, 1) -> (0, 1] for log.
+  Vec u2=Vec::loadu(data+Vec::size());
+  Vec radius=(minus_two * u1.log());//std::sqrt(-2 * std::log(u1));
+  radius=radius.sqrt();
+  Vec theta=two_pi * u2;
+  Vec output_vec=radius * theta.cos() * std + mean;
+  Vec output_vec2=radius * theta.sin() * std + mean;
```

Comment:
Can we leverage the optimization in `sincos256_ps` instead of calling cos and sin separately?

Reply:
sincos256_ps is defined in file [https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/cpu/avx_mathfun.h](url) under CPU_CAPABILITY_AVX2 flag. As we were revising function for general vectorized implementation did not use sincos256_ps.

Reply:
Would this introduce perf regression for AVX2?

Reply:
Yes. It will.

Reply:
Either abstracting sincos for all archs (do optimal impl if possible like AVX2 otherwise fallback to sin + cos, later on others can add optimal impl later), or keeping AVX2 and refactor later.

Reply:
yes. Kept AVX2  can refactor it later.